### DISCUSSION TITLE
Document base 8 requirement for `mode` in various filesystem operations?

### DISCUSSION THREAD

**Discussion by adigitoleo** [libuv/libuv#4429] [General]
I come here by way of NeoVim's bundled libuv wrapper. I had no idea that some of the filesystem functions e.g. `fs_mkdir` expected an octal `mode`, and only found this thanks to a couple of uses of `fs_mkdir` in the NeoVim implementation itself. I assume this is not something unique to the NeoVim lua wrapper (luv)? In that case, I think that should be documented somewhere. The documentation only points to `mkdir(2)` which defers to `mode_t(3)`, which only says "It is an integer type".

I guess this is also the case for some other functions e.g. `fs_open`, `fs_chmod`, ...?

**Comment by saghul** [comment]
It's the general way to express file modes on Unix, so we use that on all platforms. It could be a bit better documented indeed.

**Comment by adigitoleo** [comment]
To clarify, I'm aware that the numeric codes for chmod are in the range `000-777` however I didn't know that it is implemented here as an actual octal (base 8) value. Is that the case in glibc chmod as well?

**Reply by saghul** [reply]
Yes.

**Reply by adigitoleo** [reply]
I see, in that case it makes more sense that this is expected. In the end it's really an issue with `mode_t(3)` I guess, but it could help to spell it out in libuv as well if you like.

**Comment by adigitoleo** [comment]
For posterity: https://github.com/luvit/luv/pull/703
